plugins/gssapi.c

```diff
@@ -1772,10 +1772,18 @@ static int gssapi_client_mech_step(void *conn_context,
 	}
 
 	/* Setup req_flags properly */
-	req_flags = GSS_C_MUTUAL_FLAG | GSS_C_SEQUENCE_FLAG;
+        if (text->mech_type == NULL) {
+            /* in SASL/GSSAPI RFC4752 requires rthe INTEG flag MUST be set to
+             * TRUE in all cases */
+	    req_flags = GSS_C_INTEG_FLAG;
+        }
 	if (params->props.max_ssf > params->external_ssf) {
 	    /* We are requesting a security layer */
-	    req_flags |= GSS_C_INTEG_FLAG;
+	    req_flags |= GSS_C_MUTUAL_FLAG | GSS_C_SEQUENCE_FLAG;
+            if (text->mech_type == &gss_spnego_oid) {
```

Comment:
I would skip the condition and just add the flag.

Reply:
ok